examples/tv-casting-app/linux/simple-app-helper.cpp

```diff
@@ -243,10 +245,17 @@ CHIP_ERROR CommandHandler(int argc, char ** argv)
                             ChipLogError(AppServer, "Invalid casting player index provided: %lu", index));
         std::shared_ptr<matter::casting::core::CastingPlayer> targetCastingPlayer = castingPlayers.at(index);
 
-        matter::casting::core::EndpointFilter desiredEndpointFilter;
-        desiredEndpointFilter.vendorId = kDesiredEndpointVendorId;
+        matter::casting::core::IdentificationDeclarationOptions idOptions;
+        idOptions.mNumTargetAppInfos          = 1;
+        idOptions.mTargetAppInfos[0].vendorId = kDesiredEndpointVendorId;
+        // Attempt Commissioner-Generated Passcode commissioning flow only if the CastingPlayer indicates support for it.
+        if (targetCastingPlayer->GetSupportsCommissionerGeneratedPasscode())
+        {
+            idOptions.mCommissionerPasscode = true;
+        }
```

Comment:
Besides checking if the commissioner supports it, shouldn't this request command now also accept a CLI argument (the command could be "cast request 0 _**cdc**_") from the user to say whether the user wants to use the commissioner generated passcode method of doing commissioning? Right now it looks like the casting app will just run CDC everytime the commissioner supports it.